### PR TITLE
Add User Profile

### PR DESCRIPTION
**Pull request by stalhaali**
Implemented:

- "Skills" section where custom skills can be added.
- "About Me" section (less than or equal 250 characters).
- "Headline" section (less than or equal 50 characters).
- Personal API route `PUT /api/v1/profiles` for updating profile.
- API route `GET /api/v1/profiles/:id` for fetching profile.

### CODE REVIEW

**Comment by andy9775** [comment]
This is going to take me some time to go through, but first comment is that we should have either a `yarn.lock` or `package-lock.json` file - not both.

**Comment by stalhaali** [comment]
> This is going to take me some time to go through, but first comment is that we should have either a `yarn.lock` or `package-lock.json` file - not both.

Fixed

File: Buddies.Web/src/components/SkillList.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import Grid from '@material-ui/core/grid';
+import Skill from './Skill';
+import type { Skillobject } from '../pages/Profiles/[pid]';
+
+/* returns a list of all the skills in a grid */
+
```
Comment (andy9775) [inline Buddies.Web/src/components/SkillList.tsx:7]:
```suggestion
/* returns a list of all the skills in a grid */
```

File: Buddies.Web/src/components/Websites.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <Grid container>
+        <Grid item xs={11}>
+          <Typography color="inherit" variant="h6" gutterBottom>
+            Websites
+          </Typography>
+        </Grid>
+        <Grid item xs={1}>
+          {logCheck
+                        && <CustomizedDialogs color="inherit" topmarg={0} />}
+        </Grid>
```
Comment (andy9775) [inline Buddies.Web/src/components/Websites.tsx]:
```suggestion
        <Grid item xs={1}>
          {logCheck && <CustomizedDialogs color="inherit" topmarg={0} />}
        </Grid>
```

File: Buddies.Web/src/components/dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <ModeEditOutlinedIcon sx={{ marginTop: topmarg, color: { color }, '&:hover': { color: '#add8e6', cursor: 'pointer' } }} onClick={handleClickOpen} />
+      {' '}
+      {/* onClick={} pop up */}
```
Comment (andy9775) [inline Buddies.Web/src/components/dialog.tsx]: More descriptive?

**Review by andy9775 — APPROVED** [review]
Just some minor nits, also, you have an empty file - `test.txt`. Otherwise LGTM!

**Comment by andy9775** [comment]
Also, no secondary reviewer?

**Review by TianYueDong — APPROVED** [review]
looks good

**Pull request merged**